client/src/components/common/boost/BoostCalibration.tsx

```diff
@@ -0,0 +1,98 @@
+import React , {useState, useCallback} from 'react';
+import {Button, Card, Form, Col} from 'react-bootstrap';
+
+export interface BoostCalibrationProps {
+    onSet: (calibValue: number) => void;
+    onReset: () => void;
+    distTravelled: number;
+  }
+
+/**
+ * Boost calibration card, displays the calibrated distance and the 
+ * distance travelled by the bike while also allowing user to set the 
+ * calibrated distance
+ * 
+ * @returns {React.Component} Component
+ */
+export default function BoostCalibration({
+    onSet, onReset, distTravelled
+}: BoostCalibrationProps) {
+    // Controls the feedback form
+    const [validated, setValidation] = useState(false);
+    
+    const [calibValue, setCalibValue] = useState(0);
+    // Holds the difference between the current distance and the calibrated distance
+    const [calibDiff, setCalibDiff] = useState(0);
```

Comment:
The calibration difference is actually published by BOOST over MQTT: https://www.notion.so/V3-MQTT-Topics-66e6715d0e1941ffaa82020e5868fbae#28cede6e40a540eabe0e04f2876cbce7

So, I reckon rather than keeping track of it a second time here, take the calibration difference as a prop so that this page can display exactly what BOOST thinks the calibrated distance is, rather than a potentially different value.

Reply:
Wait so Boost publishes both the normal distance and the calibrated distance?

Reply:
Well the rear WM publishes the "normal" distance (the distance the bike has travelled), and BOOST publishes the calibrated distance which is just that same distance travelled plus a constant. This component is for telling BOOST what that constant should be.

Reply:
Right, I should have actually known this, I've looked into the relevant files that do this 😂